packages/components/src/external-link/index.tsx

```diff
@@ -32,12 +32,22 @@ function UnforwardedExternalLink(
 		compact( [ ...rel.split( ' ' ), 'external', 'noreferrer', 'noopener' ] )
 	).join( ' ' );
 	const classes = classnames( 'components-external-link', className );
+	/* Anchor links are percieved as external links.
+	This constant helps check for on page anchor links,
+	to prevent them from being opened in the editor. */
+	const shouldPreventDefault = href.startsWith( '#' );
```

Comment:
Can we guarantee `href` is always a `String`? For example is it possible for it to be `undefined`? If it is then we should guard here and also coerce to `Boolean`.

```suggestion
	const shouldPreventDefault = !! href?.startsWith( '#' );
```

Reply:
I am not sure how to backtrace it to see if it can be undefined for some reason, so I believe it is easier to add the check :)

Reply:
Though the type for `href` is set to `string` in types.ts. I think @walbo would be able to answer how to best handle the check.